logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/GetOneToOneConversationUseCase.kt

```diff
@@ -4,13 +4,21 @@ import com.wire.kalium.logic.data.conversation.Conversation
 import com.wire.kalium.logic.data.conversation.ConversationRepository
 import com.wire.kalium.logic.data.user.UserId
 import com.wire.kalium.logic.functional.fold
+import com.wire.kalium.util.KaliumDispatcher
+import com.wire.kalium.util.KaliumDispatcherImpl
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.map
 
 class GetOneToOneConversationUseCase internal constructor(
-    private val conversationRepository: ConversationRepository
+    private val conversationRepository: ConversationRepository,
+    private val dispatchers: KaliumDispatcher = KaliumDispatcherImpl
 ) {
 
-    suspend operator fun invoke(otherUserId: UserId): Result = conversationRepository.getOneToOneConversationWithOtherUser(otherUserId)
-        .fold({ Result.Failure }, { Result.Success(it) })
+    suspend operator fun invoke(otherUserId: UserId): Flow<Result> =
```

Comment:
Let's take advantage of these changes, and we can add a small block of documentation, for the purpose of the Use case :D